mypyc/genops.py

```diff
@@ -1067,6 +1095,11 @@ def for_loop_helper(self, index: Lvalue, expr: Expression,
             body_insts()
             self.add(Goto(increment_block))
 
+            if else_insts is not None:
+                self.activate_block(else_block)
+                else_insts()
+                self.add(Goto(exit_block))
```

Comment:
The else blocks are being generated before the loop_stack is popped, which is wrong, since it will result in break and continue in an else block going to the wrong place.

Moreover I think we can arrange to only have this code repeated once, after all of the cases. I think we can also avoid doing an error check after a `break`, which we do currently but don't need to, and I think will fall out from the tweaks needed to not duplicate the else code.